src/main/java/Bob/util/controller/LoadSave.java

```diff
@@ -0,0 +1,238 @@
+     * @throws BobInvalidLoadValue when type of task is found to be invalid.
+     */
+    public static Task createTask(String type, boolean status, String details) throws BobInvalidLoadValue {
+        Task task;
+        String[] tokenDetails;
+        switch (type) {
+        case "T":
+            task = new ToDos(details, status);
+            break;
+        case "D":
+            tokenDetails = getTokenDetails(type, details);
+            LocalDate deadline = Parser.parseDeadline(tokenDetails[1]);
+            task = new Deadlines(tokenDetails[0], status, deadline);
+            break;
+        case "E":
+            tokenDetails = getTokenDetails(type, details);
+            LocalDate[] eventDates = Parser.parseEventDate(tokenDetails[1]);
+            task = new Events(tokenDetails[0], eventDates[0], eventDates[1]);
+            break;
+        default:
+            throw new BobInvalidLoadValue();
+        }
+        return task;
+    }
+
+    /**
+     * Creates and appends a task to a list.
+     *
+     * @param list a Task class list.
+     * @param data the data of the task to be created and appended.
+     * @throws BobInvalidLoadLength when the length of the data does not meet minimum requirements.
+     * @throws BobInvalidLoadValue when data have unexpected arguments.
+     */
+    public static void addToList(ArrayList<Task> list, String data) throws BobInvalidLoadValue, BobInvalidLoadLength {
+        if (data.length() < MIN_TASK_LENGTH) {
+            throw new BobInvalidLoadLength();
+        }
+        try {
+            String taskType = data.substring(TASK_INDICATOR, TASK_INDICATOR_END);
+            boolean taskStatus = getTaskStatus(data.substring(STATUS_INDICATOR, STATUS_INDICATOR_END));
+            String taskDetails = data.substring(TASK_DETAILS);
+            Task task = createTask(taskType, taskStatus, taskDetails);
+            list.add(task);
+        } catch (BobInvalidLoadValue e) {
+            printLoadError(MESSAGE_INVALID_VALUES);
+            throw new BobInvalidLoadValue();
+        }
+    }
+
+    /**
+     * Reads data from a file and restores them into the list.
+     *
+     * @param list a Task class list to be restored.
+     * @throws FileNotFoundException when the file cannot be found.
+     */
+    public static void readDataFromFile(ArrayList<Task> list) throws FileNotFoundException {
+        try {
+            File file = new File(FILEPATH);
+            Scanner in = new Scanner(file);
+            while (in.hasNext()) {
+                if (list.size() >= Command.MAX_TASK_ID) {
+                    printLoadError(MESSAGE_SAVE_DATA_EXCEEDED);
+                    break;
+                }
+                addToList(list, in.nextLine());
+            }
+        } catch (BobInvalidLoadLength e) {
+            printLoadError(MESSAGE_INVALID_LENGTH);
+        } catch (BobInvalidLoadValue e) {
+            // already handled, catch to stop loop
```

Comment:
You could ensure that the comment is of [same level of indentation](https://se-education.org/guides/conventions/java/basic.html#comments) as code.